js/router.js

```diff
@@ -149,10 +166,71 @@ export default class ObRouter extends Router {
     app.pageNav.setAddressBar(displayRoute);
   }
 
-  execute(callback, args) {
-    app.loadingModal.open();
+  execute(callback, args, name, options = {}) {
+    if (this.closeUnconfirmedRollBack) {
+      this.closeUnconfirmedRollBack = false;
+      return false;
+    }
+
     this.navigate(this.standardizedRoute(), { replace: true });
 
+    // We'll iterate through any open modal which have a confirmClose method
```

Comment:
The router work is where the bulk of the complexities lies with this feature. It's one thing to put up a confirm when you click close, but ideally we also want it if the user navigates away from the modal as well. Additional complexity comes into play with the fact that you could have multiple modals open that need confirmation (e.g. you may have begun editing a listing and then decided to create a listing on top of the one you are editing).